### PR TITLE
feat: split `rustywind` into multiple crates so that it can also be made available as a library

### PR DESCRIPTION
**Pull request by Rolv-Apneseth**
In relation to #95 

Continuing on from [bram209's branch](https://github.com/bram209/rustywind/tree/multiple-crates), so most of the credit goes to him.

Opening this PR now because I need some pointers:

1. That latest commit I pushed, was just replacing the default `WriteMode::DryRun` with `WriteMode::ToConsole` because `clippy` is complaining that `ToConsole` is never constructed, and `DryRun` already has an option dedicated to it. If this is wrong, would you mind clarifying what the purpose of `ToConsole` is, and how I should fix that linting error?

2. For adding a workflow to publish `rustywind_core`, would that just be something like this (found on the internet):

    ```yml
    on:
      push:
        tags:        
          - '*'
      workflow_dispatch:

    name: Publish Library

    jobs:
      publish:
        name: Publish
        runs-on: ubuntu-latest
        steps:
          - name: Checkout sources
            uses: actions/checkout@v2

          - name: Install stable toolchain
            uses: actions-rs/toolchain@v1
            with:
              profile: minimal
              toolchain: stable
              override: true

          - run: cargo publish --package rustywind_core --token ${CRATES_TOKEN}
            env:
              CRATES_TOKEN: ${{ secrets.CRATES_TOKEN }}
    ```

    or should it be using an action like [this](https://github.com/katyo/publish-crates)?
    
3. For `rustywind_vite`, what should the description be? And is this also to be published as a separate library?

4. Should versioning be different for all 3 crates?

5. How can I check CI/CD workflows are working? I have ensured `clippy` and `rustfmt` are happy locally.
    
Did I miss anything else?

### CODE REVIEW

**Comment by praveenperera** [comment]
Hey @Rolv-Apneseth sorry I didn’t see this till now. I won’t be able to review till Tuesday. 

1. That’s good console as default makes sense
2. I don’t need this for now I’ll do releases manually 
3. Yes let’s just publish as another package, the point it to make rustywind option to automatically get the class order from the output css file to a work with vite. 
4. Yes
5. Running now. 

Is this ready to review now? Or does it need to be in draft longer?

**Comment by Rolv-Apneseth** [comment]
Sure, no rush at all.

Sorry, wanted to ask those questions first, it can be ready for review yeah. Just need to change the versions and look into that workflow failing

**Comment by Rolv-Apneseth** [comment]
Not sure what the issue was with that workflow failing, globbing was broken somehow? I could maybe change it to just run `cargo fmt --check -- --edition 2021`

**Comment by praveenperera** [comment]
> Not sure what the issue was with that workflow failing, globbing was broken somehow? I could maybe change it to just run `cargo fmt --check -- --edition 2021`

Ya let’s change it to that

**Comment by Rolv-Apneseth** [comment]
Cool I'll change it. Is this comment still relevant?

```
# When pushed to master, run `cargo +nightly fmt --all` and open a PR.
```

**Comment by Rolv-Apneseth** [comment]
Oh, or [this one from dtolnay](https://github.com/dtolnay/rust-toolchain) if you prefer

**Comment by praveenperera** [comment]
No that can be removed

**Comment by Rolv-Apneseth** [comment]
> No that can be removed

Alright done. Any thoughts about changing the action or leave it as is?

**Review by praveenperera — APPROVED** [review]

**Comment by praveenperera** [comment]
looks good @Rolv-Apneseth thanks! I'll merge it in today.

File: .github/workflows/rustfmt.yml [hunk 1]
```diff
@@ -15,4 +14,4 @@ jobs:
           override: true
           components: rustfmt
 
-      - run: rustfmt --edition 2021 **/*.rs --check
+      - run: cargo fmt --check -- --edition 2021
```
Comment (praveenperera) [inline .github/workflows/rustfmt.yml:17]:
```suggestion
      - run: cargo fmt --check
```

**Comment by Rolv-Apneseth** [comment]
Nice, thank you

**Comment by praveenperera** [comment]
Added some docs: https://docs.rs/rustywind_core/latest/rustywind_core/

**Pull request merged**